lisa/tests/scheduler/misfit.py

```diff
@@ -295,6 +295,7 @@ def _test_cpus_busy(self, task_state_dfs, cpus, allowed_idle_time_s):
         return res
 
     @TasksAnalysis.df_task_states.used_events
+    @check_noisy_tasks(1)
```

Comment:
Hmm that's not very intuitive way to check for noisy tasks. Not for me at least - I am not a big python user in general so I could be the exception here.

What does this do? Why a simple function call from within this function isn't a better alternative?

Reply:
In short, the `check_noisy_tasks` decorator turns a call to `func()` into:

```
# This is the function to which that decorator is attached to
res = func(<right parameters dumped here>)

pid, comm, duration = self.find_noisiest_task()
ok = 100 * duration / self.trace.time_range < threshold_pct
add_relevant_metrics(res)

if not ok:
    res.result = Result.UNDECIDED

return res
```

The good thing is it prevents us from having to add that `noisy_task_pct` threshold parameter to every single test method that we want to guard against noisy tasks, and calling the right function just before returning.

But that's also a bad thing - it means we can't easily re-run that test offline while playing with the noise threshold like the other test thresholds.

ATM it's in the form of a decorator because it was the least intrusive way and that made for a simpler RFC, I'll have to think more about it :)

Reply:
Cool. Thanks for explanation. If you ended up staying in this route - please add this info in the commit message. It's an important design decision that is worth explaining in the commit message for whoever comes up with a reason to need to change this in the future.

By the way, if the function has a named parameter with a default value - would this go around the need to pass the threshold with every call? Using a function will make for a simpler code if it is as convenience as this approach.

Reply:
The main thing the decorator offers is that it can intercept the return value of the original function and update it the way it wants. An alternative way would be to "hijack" explicitly the return value like that:
```
def mytest(..., threshold_pct=42):
    ...
    res = ResultBundle()
    ...
    return check_noisy_tasks(res, threshold_pct)
```
This way exposes `threshold_pct` to the user, the decorator way allows getting the feature without even looking for `return` in the function. Either ways are good to me here, as we don't need to update the function itself (unlike for `requires_events`, where we need to set attributes to be picked up by the documentation).